File: examples/python/schemas-html.py
from meshagent.api.schema import MeshSchema, ElementType, ChildProperty, ValueProperty

schema = MeshSchema(
    root_tag_name="html",
    elements=[
        ElementType(
            tag_name="html",
            properties=[
                # a ChildProperty describes the type of children that an element allows. There can at most one child
                # property for each element type, but the child property can allow multiple types of child elements
                ChildProperty(name="children", child_tag_names=["body"])
            ]),
        
        ElementType(
            tag_name="body",
            properties=[
                # Our body can only contain paragraph elements
                ChildProperty(name="children", child_tag_names=["p"])
            ]),
        
        ElementType(
            tag_name="p",
            properties=[
                # A ValueProperty property describes an attribute that contains a single value
                ValueProperty(name="class", type="string"),
            ]),
    ]
)

File: examples/dart/schemas-html.dart
import 'package:meshagent/meshagent.dart';

final schema = MeshSchema(
  rootTagName: 'html',
  elements: [
    ElementType(
      tagName: 'html',
      properties: [
        // A ChildProperty describes the type of children that
        // an element allows. There can be at most one child
        // property for each element type, but the child property
        // can allow multiple types of child elements.
        ChildProperty(
          name: 'children',
          childTagNames: ['body'],
        ),
      ],
    ),
    ElementType(
      tagName: 'body',
      properties: [
        // Our body can only contain paragraph elements.
        ChildProperty(
          name: 'children',
          childTagNames: ['p'],
        ),
      ],
    ),
    ElementType(
      tagName: 'p',
      properties: [
        // A ValueProperty describes an attribute that
        // contains a single value.
        ValueProperty(
          name: 'class',
          type: 'string',
        ),
      ],
    ),
  ],
);

void main() {
  // Here you could do something with `schema`, like
  // passing it to a registry or printing it out.
  print(schema);
}
